packages/grid/x-data-grid-pro/src/DataGridPro/useDataGridProComponent.tsx

```diff
@@ -81,10 +85,10 @@ export const useDataGridProComponent = (
   useGridInitializeState(columnPinningStateInitializer, apiRef, props);
   useGridInitializeState(rowGroupingStateInitializer, apiRef, props);
   useGridInitializeState(columnsStateInitializer, apiRef, props);
-  useGridRowGrouping(apiRef, props); // FIXME Needs to be called before the rows state initialization because it registers a rows group builder
```

Comment:
I moved the strategy processors into `useGridTreeDataPreProcessors` and `useGridRowGroupingPreProccessors` so we can now respect the clean order